rust/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -127,34 +128,38 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
         let (ism_addresses, threshold) = ism.modules_and_threshold(message).await.context(CTX)?;
         let threshold = threshold as usize;
 
-        let metas = join_all(
-            ism_addresses
-                .iter()
-                .map(|ism_address| self.base.build(*ism_address, message)),
-        )
-        .await;
-
-        let sub_modules = join_all(
+        let sub_modules_and_metas = join_all(
             ism_addresses
                 .iter()
-                .map(|ism_address| self.base.build_ism(*ism_address)),
+                .map(|ism_address| self.base.build_ism_and_metadata(*ism_address, message)),
```

Comment:
Was thinking you could add ism_address here, like `map(|ism_address|) (*ism_address, self.base.build_ism_and_meta(/* ... */)))`

Edit: but this is wrong